eZ/Publish/Core/REST/Server/Controller/Location.php

```diff
@@ -247,12 +247,17 @@ public function moveSubtree($locationPath, Request $request)
                 // Trash the subtree
                 $trashItem = $this->trashService->trash($locationToMove);
 
-                return new Values\ResourceCreated(
-                    $this->router->generate(
-                        'ezpublish_rest_loadTrashItem',
-                        array('trashItemId' => $trashItem->id)
-                    )
-                );
+                if (isset($trashItem)) {
+                    return new Values\ResourceCreated(
+                        $this->router->generate(
+                            'ezpublish_rest_loadTrashItem',
+                            array('trashItemId' => $trashItem->id)
+                        )
+                    );
+                } else {
```

Comment:
You could ditch the `else`, and move the code one level up, since the above if block will return if true.